File: mod_02_logica_e_algoritmos/imc/python_imc_oo/PessoaImc.py
from Imc import Imc

class PessoaImc(Imc):
    def __init__(self, nome, idade, peso, altura):
        super().__init__(peso, altura)
        self.nome = nome
        self.idade = idade

    def classificacao(self):
        imc = self.getImc()
        if imc < 18.5:
            return "Abaixo do peso"
        elif 18.5 <= imc < 24.9:
            return "Peso normal"
        elif 25 <= imc < 29.9:
            return "Sobrepeso"
        elif 30 <= imc < 34.9:
            return "Obesidade grau I"
        elif 35 <= imc < 39.9:
            return "Obesidade grau II"
        else:
            return "Obesidade grau III (obesidade mórbida)"

File: mod_02_logica_e_algoritmos/imc/java_imc_oo/src/PessoaImc.java
public class PessoaImc extends Imc {
    private String nome;
    private int idade;

    public PessoaImc(String nome, int idade, double peso, double altura) {
        super(peso, altura);
        this.nome = nome;
        this.idade = idade;
    }

    @Override
    public String classificacao() {
        double imc = getImc();
        if (imc < 18.5) {
            return "Abaixo do peso";
        } else if (imc >= 18.5 && imc < 24.9) {
            return "Peso normal";
        } else if (imc >= 25 && imc < 29.9) {
            return "Sobrepeso";
        } else if (imc >= 30 && imc < 34.9) {
            return "Obesidade grau I";
        } else if (imc >= 35 && imc < 39.9) {
            return "Obesidade grau II";
        } else {
            return "Obesidade grau III (obesidade mórbida)";
        }
    }
}
